OsmAnd/src/net/osmand/plus/measurementtool/OptionsBottomSheetDialogFragment.java

```diff
@@ -54,7 +54,7 @@ public void onClick(View v) {
 
 		items.add(new DividerHalfItem(getContext()));
 
-		if (addLineMode) {
+		if (addLineMode && !snapToRoadEnabled) {
```

Comment:
We have Save button at top-right corner of the screen. Did you change it behaviour as well? Is it needed?

Reply:
changed behavior, now it opens "Save as new GPX file" dialog